### PR TITLE
E2E Selectors: Add 'modified' dist-tag when release includes actual changes

### PR DESCRIPTION
**Pull request by sunker**
**What is this feature?**

This PR adds a `modified` dist-tag to a release of @grafana/e2e-selectors package **only** if the release includes changes, i.e that any of the files in the package has been modified. 

**Why do we need this feature?**

Plugin E2E will need to bump the e2e-selectors dependency when there's changes to the selectors. However, a new canary release of the e2e-selectors package will published every time a PR in this repo is merged to main. With the new `modified` flag, consumers of this package can choose to only bump the dep when a release actually has real changes. 

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: scripts/publish-npm-packages.sh [hunk 1]
```diff
@@ -44,3 +44,17 @@ echo "$registry_without_protocol/:_authToken=${NPM_TOKEN}" >> ~/.npmrc
 for file in ./npm-artifacts/*.tgz; do
     npm publish "$file" --tag "$dist_tag" --registry "$registry"
 done
+
+# Check if any files in packages/grafana-e2e-selectors were changed. If so, add a 'modified' tag to the package
+COUNT=$(git diff HEAD~1..HEAD --name-only -- packages/grafana-e2e-selectors | awk 'END{print NR}')
```
Comment (leventebalogh) [inline scripts/publish-npm-packages.sh:49]: Nit: maybe `CHANGES_COUNT` would be slightly better name, not too important though.

File: scripts/publish-npm-packages.sh [hunk 2]
```diff
@@ -44,3 +44,17 @@ echo "$registry_without_protocol/:_authToken=${NPM_TOKEN}" >> ~/.npmrc
...
+if (( $COUNT > 0 )); then
+    # Wait a little bit to allow the package to be published to the registry
+    sleep 5s
+    regex_pattern="canary: ([0-9.-]+)"
+    TAGS=$(npm dist-tag ls @grafana/e2e-selectors)
+    if [[ $TAGS =~ $regex_pattern ]]; then
+        echo "$COUNT file(s) in packages/grafana-e2e-selectors were changed. Adding 'modified' tag to @grafana/e2e-selectors@${BASH_REMATCH[1]}"
+        npm dist-tag add @grafana/e2e-selectors@${BASH_REMATCH[1]} modified
```
Comment (leventebalogh) [inline scripts/publish-npm-packages.sh:57]: I was wondering if a dist-tag like `alpha` would suite better? Just an idea.

**Review by leventebalogh — APPROVED** [review]
LGTM, left minor comments. Let's also wait for what @jackw thinks.

**Review by tolzhabayev — APPROVED** [review]

File: scripts/publish-npm-packages.sh (hunk 2 above)
Comment (sunker) [reply scripts/publish-npm-packages.sh:57]: That's a good option too, thanks! Once the enterprise datasources stop using the canary releases on the e2e selectors in their ci, my hope is that we can remove this new dist tag and simply _only_ release this package once its actually changed.

**Pull request merged**